cli/src/pcluster/api/controllers/cluster_operations_controller.py

```diff
@@ -305,16 +315,81 @@ def update_cluster(
+    if client_token:
+        raise BadRequestException("clientToken is currently not supported for this operation")
+    cluster_config = read_config(update_cluster_request_content.cluster_configuration)
+
+    try:
+        cluster = Cluster(cluster_name)
+        if not check_cluster_version(cluster, exact_match=True):
+            raise BadRequestException(
+                f"the update can be performed only with the same ParallelCluster version ({cluster.stack.version}) "
+                "used to create the cluster."
+            )
+
+        if dryrun:
+            cluster.validate_update_request(
+                target_source_config=cluster_config,
+                force=force_update,
+                validator_suppressors=get_validator_suppressors(suppress_validators),
+                validation_failure_level=FailureLevel[validation_failure_level],
+            )
+            raise DryrunOperationException()
+
+        changes, ignored_validation_failures = cluster.update(
+            target_source_config=cluster_config,
+            validator_suppressors=get_validator_suppressors(suppress_validators),
+            validation_failure_level=FailureLevel[validation_failure_level],
+            force=force_update,
         )
+
+        return UpdateClusterResponseContent(
+            cluster=ClusterInfoSummary(
+                cluster_name=cluster_name,
+                cloudformation_stack_status=CloudFormationStatus.UPDATE_IN_PROGRESS,
+                cloudformation_stack_arn=cluster.stack.id,
+                region=os.environ.get("AWS_DEFAULT_REGION"),
+                version=cluster.stack.version,
+                cluster_status=cloud_formation_status_to_cluster_status(CloudFormationStatus.UPDATE_IN_PROGRESS),
+            ),
+            validation_messages=validation_results_to_config_validation_errors(ignored_validation_failures) or None,
+            change_set=_get_changeset(changes),
+        )
+    except ConfigValidationError as e:
+        raise _handle_config_validation_error(e)
+    except ClusterUpdateError as e:
+        raise _handle_cluster_update_error(e)
+    except (NotExistsClusterActionError, StackNotFoundError):
+        raise NotFoundException(
+            f"cluster '{cluster_name}' does not exist or belongs to an incompatible ParallelCluster major version."
+        )
+
+
+def _get_changeset(changes):
+    if changes is None or len(changes) <= 1:
+        return None
+
+    key_indexes = {key: index for index, key in enumerate(changes[0])}
+    return [
+        Change(
+            parameter=row[key_indexes["parameter"]],
+            requested_value=row[key_indexes["new value"]],
+            current_value=row[key_indexes["old value"]],
+        )
+        for row in changes[1:]
+    ]
+
+
+def _handle_cluster_update_error(e):
+    return UpdateClusterBadRequestException(
+        UpdateClusterBadRequestExceptionResponseContent(message=str(e), change_set=_get_changeset(e.update_changes))
```

Comment:
I think we are missing `updateValidationErrors` and `configurationValidationErrors` in this response

Reply:
Added `updateValidationErrors` for issues with the patch. Note that when we run the check on the patch, the configuration has already been successfully validated. We do return the suppressed validation errors with the response in the case of a successful call, however since for patch validation failures we would throw a BadRequest. I was wondering whether adding the suppressed config validation errors to the exception thrown by the patch validation would be confusing (as the suppressed errors were not the cause of the BadRequest failure).